feat: Add ability to configure emptyDir params (sizeLimit, medium)

mkilchhofer opened the pull request:
**Is this a bug fix or adding new feature?**

It adds a new feature

**What is this PR about? / Why do we need it?**

We use both the EBS and the EFS CSI drivers. The EBS driver offers this capability already:
- https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1840
- https://github.com/kubernetes-sigs/aws-ebs-csi-driver/commit/3cc74df677e6ae7ac4a0c105b58cba7edb191b8f#diff-38890fc178a6bb362418969605e8650e75296dedaf1eef29dc3b21400ff36676L220

**What testing is done?** 

```bash
$ helm template aws-efs-csi-driver .
(..)
          securityContext:
            allowPrivilegeEscalation: false
            readOnlyRootFilesystem: true
      volumes:
        - name: socket-dir
          emptyDir: {}     # <-- see here
---
# Source: aws-efs-csi-driver/templates/csidriver.yaml
apiVersion: storage.k8s.io/v1
(..)
```

And with customized emptyDir:

```bash
$ helm template aws-efs-csi-driver . --set controller.socketDirVolume.emptyDir.sizeLimit=100Mi
(..)
          securityContext:
            allowPrivilegeEscalation: false
            readOnlyRootFilesystem: true
      volumes:
        - name: socket-dir
          emptyDir:           # <-- see here
            sizeLimit: 100Mi  # 
---
# Source: aws-efs-csi-driver/templates/csidriver.yaml
apiVersion: storage.k8s.io/v1
(..)
```

mkilchhofer commented:
Hi @dankova22 

Friendly reminder that I wait on a review for this PR :)

DavidXU12345 commented on charts/aws-efs-csi-driver/templates/controller-deployment.yaml:
```diff
@@ -180,7 +180,11 @@ spec:
       volumes:
         - name: socket-dir
+          {{- if .Values.controller.socketDirVolume }}
+          {{- toYaml .Values.controller.socketDirVolume | nindent 10 }}
```
I think you can just keep this line and remove the if-else conditions since the default value in `values.yaml` is already `emptyDir: {}`

i.e.

```
      volumes:
        - name: socket-dir
        {{- toYaml .Values.controller.socketDirVolume | nindent 10 }}
        {{- with .Values.controller.volumes }}
        {{- toYaml . | nindent 8 }}
        {{- end }}
```

DavidXU12345 replied:
Actually I think it is safer to add the guard logic just in case other customers are using outdated `values.yaml`. It is good to merge after you fix the typo in the commit message, thanks

mkilchhofer commented:
> Also, could you please fix the typo for `medium` in the commit message?

Hi @DavidXU12345,
Thanks for your review. I addressed it and also rebased against latest `master`.
